apps/web/src/libs/wallet-connect-adapter.ts

```diff
@@ -86,24 +86,34 @@ export class WalletConnectAdapter extends EventEmitter implements IConnector {
   }
 
   public async signTransaction(transaction: Transaction, message?: string) {
-    if (!this._connector) throw new Error('WalletConnect not initialized')
+    try {
```

Comment:
The try/catch allows catching a submitted transaction that was declined. Now the error screen will appear in this case. The user can select to Try Again.